### PR TITLE
Docs: various optimizations and fixes 

### PR DESCRIPTION
**Pull request by Maroc-OS**
fixes for problem:

```
INFO    -  Doc file 'index.md' contains an unrecognized relative link 'nn/#tinygrad.nn.state.get_parameters', it was left as is. Did you mean 'nn.md#tinygrad.nn.state.get_parameters'?
INFO    -  Doc file 'index.md' contains an unrecognized relative link 'tensor/#tinygrad.Tensor.conv2d', it was left as is. Did you mean 'tensor.md#tinygrad.Tensor.conv2d'?
INFO    -  Doc file 'index.md' contains an unrecognized relative link 'tensor/#tinygrad.Tensor.sparse_categorical_crossentropy', it was left as is. Did you mean 'tensor.md#tinygrad.Tensor.sparse_categorical_crossentropy'?
INFO    -  Doc file 'index.md' contains an unrecognized relative link 'nn/#tinygrad.nn.Conv2d', it was left as is. Did you mean 'nn.md#tinygrad.nn.Conv2d'?
INFO    -  Doc file 'index.md' contains an unrecognized relative link 'tensor/#tinygrad.Tensor.realize', it was left as is. Did you mean 'tensor.md#tinygrad.Tensor.realize'?
```

Allow highlighted func/class types in table of contents:
Screen Shot 2024-04-18 at 22 12 44

Add repo to docsite:
Screen Shot 2024-04-18 at 22 05 39

Small clean up in markdown warnings.

Add a better message to home page.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
-Welcome to the docs for tinygrad. This page is for users of the tinygrad library. We also have [developer docs](developer.md)
+# Home
+
+Welcome to the tinygrad documentation! Here, you will find all the information you need to use the tinygrad library effectively. For details on how tinygrad is developed and maintained, check out the [developer documentation](developer.md). Enjoy exploring the docs!
```
Comment (chenyuxyz) [inline docs/index.md:3]: can you revert this change? formatting fix is fine but i don't like this content change

**Comment by geohot** [comment]
"All docs and whitespace changes will be closed unless you are a well-known contributor. The people writing the docs should be those who know the codebase the absolute best."
